src/CloudNative.CloudEvents.Amqp/AmqpClientExtensions.cs

```diff
@@ -85,6 +84,17 @@ public static class AmqpClientExtensions
                         cloudEvent[attributeName] = property.Value;
                     }
                 }
+                // Populate the data after the rest of the CloudEvent
+                if (message.BodySection is Data data)
+                {
+                    // TODO: Will data.Binary always be populated? Perhaps it'll be data.Buffer?
+                    formatter.DecodeBinaryModeEventData(data.Binary, cloudEvent);
```

Comment:
Looks like it will - https://github.com/Azure/amqpnetlite/blob/master/src/Framing/Data.cs#L46-L50

Reply:
That's good to know, thanks. (I'll change the TODO to explain that.)